### DISCUSSION TITLE
Validation in body and params

### DISCUSSION THREAD

**Discussion by Subin-Qreative** [spec-first/connexion#1803] [Q&A]
Hi is there any example of validation on the params in body/payload and in parameters.
Currently there is type validation i am interested in the content of the paylaod itself e.g. Temperate , if the value is above 50 reject the api call outright saying invalid range or something like that. I saw validation section in the documentation but I couldn't get my head around it. 

Thanks

**Comment by RobbeSneyders** [comment]
Hi @Subin-Qreative, Connexion validates the payload based on your OpenAPI specification. If you specify these limitations in your OpenAPI spec, Connexion will enforce them. For more info, refer to the [validation docs](https://connexion.readthedocs.io/en/latest/validation.html).

**Comment by sumasagrp** [comment]
So if i'm understanding this right ,  i can use the https://github.com/python-jsonschema/jsonschema 
to validate by changing the content section in the yaml file? 

          content:
            text/plain:
              schema:
                type: string
                example: "hello dave!"

also  are there any additional properties  e.g. "example:" that is connexion specific,
 ? I couldn't find this particular one in the jsonschema docs.

Thanks

**Reply by RobbeSneyders** [reply]
You do not need to use `jsonschema` yourself. If you define the content section in your OpenAPI specification, Connexion will validate incoming requests for you.

**Comment by sumasagrp** [comment]
Found this actually which answered my question:
https://swagger.io/docs/specification/data-models/keywords/
